src/Resources/Pieces.php

```diff
@@ -30,29 +27,35 @@ public function __construct(array $attributes = [])
      * @param  \Mvdnbrk\DhlParcel\Resources\Piece|array  $value
      * @return void
      */
-    public function add($value)
+    public function add($value): void
     {
         if ($value instanceof Piece) {
             $this->items[] = $value;
 
             return;
         }
 
-        $this->items[] = new Piece($value);
+        if (is_array($value)) {
+            $this->items[] = new Piece($value);
+
+            return;
+        }
+
+        throw ResourceNotAccepted::forResource($value);
```

Comment:
Do not throw an exception here.

Reply:
I don't see a reason to not throw an exception here.